elixir/apps/web/lib/web/live/sites/resources/edit.ex

```diff
@@ -54,7 +59,8 @@ defmodule Web.Sites.Resources.Edit do
               required
             />
 
-            <.filters_form form={@form[:filters]} />
+            <.filters_form :if={@traffic_filters_enabled?} form={@form[:filters]} />
+            <.beta_filters_form :if={!@traffic_filters_enabled?} form={@form[:filters]} />
```

Comment:
Would it be better if instead of creating a new component we just use `:if={Config.traffic_filters_enabled?()}` inside the implementation of `<.filters_form />` component? We don't even need to assign it and pass down the chain of components.

Reply:
So, I did it this way because it seemed very simple to rip out later.  I did consider updating the `filters_form` component, but after looking at it, it felt like the component had enough going on inside of it as is.  

With the current code, ripping it out will be as easy as deleting that one line, removing the `:if` check in the line above it, and then deleting an entire function in the `components.ex` file and there would be no chance of messing up any of the currently built logic that exists in the component we want to keep long term.

Also, regarding: 
> We don't even need to assign it and pass down the chain of components.

How would the `filters_form` component get the `@traffic_filters_enabled?` var if we didn't pass it down through `attr` ?

Reply:
> So, I did it this way because it seemed very simple to rip out later. I did consider updating the filters_form component, but after looking at it, it felt like the component had enough going on inside of it as is.

Hm, I'm touching this code right now and solving a similar problem. I will push a PR so we can discuss which approach looks cleaner.

> How would the filters_form component get the @traffic_filters_enabled? var if we didn't pass it down through attr ?

You can always call a function directly: `:if={Config.traffic_filters_enabled?()}`

Reply:
What happens if we just hide the traffic filters form without putting hidden elements? It's not required in the changeset anyways.

Reply:
So, technically we could do that, but then when we flipped the feature on, all the resources that people created during the time the feature was off would then not have any traffic filtering rules and we'd end up blocking all of the traffic to those resources, because there are no rules in place.

Reply:
I think right now empty list means "allow all"

Reply:
Right now it might, but at some point it's not going to correct?  Otherwise, why even have "allow all"?

Reply:
Hm, I was thinking that it will be always like that: no filters - all traffic allowed, with filters - only whitelisted one. But maybe my logic is not correct here 🤷‍♂️, wdyt?

Reply:
Well, if that's how we want it to work, then in my opinion we shouldn't put in the option: `Permit All` in the traffic filter form, because then you have 2 states that mean the same thing.

From my perspective, us having the `Permit All` option meant that if you did not choose any of the options, the you were not allowing any traffic to that given resource.  That list of options was an 'Allow' list and if your 'Allow' list was empty then you don't allow anything.

If we go with the idea of "an empty list means allow all" and we wanted to give the user the ability to block all traffic to a given resource, then we'd need to add an option saying `Block All`, but now the list of options is a mix between "Allow" options and "Deny" options.  And if we didn't add that option, then there would be no way for a user to block all traffic to a given resource.

My vote would be to keep the `Permit All` option and the traffic filter list is an 'Allow' list, which would mean an empty list would block all traffic.  That said,  I'm not set in stone on this opinion, so I'm interested to hear what your (and everyone else) thoughts are.

cc: @jamilbk @jefferenced @conectado

Reply:
cc @rbhitchcock